extensions/http/src/main/java/org/mule/extension/http/api/request/HttpRequesterOperations.java

```diff
@@ -0,0 +1,165 @@
+ * LICENSE.txt file.
+ */
+package org.mule.extension.http.api.request;
+
+import static java.lang.Integer.MAX_VALUE;
+import org.mule.extension.http.api.HttpConnector;
+import org.mule.extension.http.api.HttpResponseAttributes;
+import org.mule.extension.http.api.HttpSendBodyMode;
+import org.mule.extension.http.api.HttpStreamingType;
+import org.mule.extension.http.api.request.builder.HttpRequesterRequestBuilder;
+import org.mule.extension.http.api.request.client.HttpClient;
+import org.mule.extension.http.api.request.validator.ResponseValidator;
+import org.mule.extension.http.api.request.validator.SuccessStatusCodeValidator;
+import org.mule.extension.http.internal.request.HttpRequester;
+import org.mule.runtime.api.message.MuleMessage;
+import org.mule.runtime.core.api.MuleEvent;
+import org.mule.runtime.core.api.MuleException;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataKeyId;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
+import org.mule.runtime.extension.api.annotation.param.Connection;
+import org.mule.runtime.extension.api.annotation.param.Optional;
+import org.mule.runtime.extension.api.annotation.param.UseConfig;
+import org.mule.runtime.module.http.api.HttpConstants;
+import org.mule.runtime.module.http.internal.HttpParser;
+
+import java.util.function.Function;
+
+
+public class HttpRequesterOperations
+{
+
+    private static final int WAIT_FOR_EVER = MAX_VALUE;
+
+    /**
+     * Consumes an HTTP service.
+     *
+     * @param path Path where the request will be sent.
+     * @param method The HTTP method for the request.
+     * @param host Host where the requests will be sent.
+     * @param port Port where the requests will be sent.
+     * @param source The expression used to obtain the body that will be sent in the request. Default is empty, so the payload will be used as the body.
+     * @param followRedirects Specifies whether to follow redirects or not.
+     * @param parseResponse Defines if the HTTP response should be parsed or it's raw contents should be propagated instead.
+     * @param requestStreamingMode Defines if the request should be sent using streaming or not.
+     * @param sendBodyMode Defines if the request should contain a body or not.
+     * @param responseTimeout Maximum time that the request element will block the execution of the flow waiting for the HTTP response.
+     * @param responseValidator Configures error handling of the response.
+     * @param config the {@link HttpConnector} configuration for this operation. All parameters not configured will be taken from it.
+     * @param muleEvent the current {@link MuleEvent}
+     * @return a {@link MuleMessage} with {@link HttpResponseAttributes}
+     */
+    @MetadataScope(keysResolver = MetadataResolver.class, outputResolver = MetadataResolver.class)
+    public MuleMessage<Object, HttpResponseAttributes> request(String path,
```

Comment:
this is ugly enough... is it actually better to have this than what you had in 3.x?